### ISSUE TITLE
V3 Suggestion: Notification Pop-Up to clear cache on new updates

### ISSUE THREAD

**Issue by GetLikha** [silexlabs/Silex#1541]
web.snapchat.com has a feature where if they have a new update, it will notify the user if there's a new update. It'll ask them to set the update by pressing a reload button on the notification. Once they click it, the page will reload and the update will be implemented.

I'm guessing upon reload, it clears the cache _for that specific site_ so the browser will be forced to fetch new files, applying the update.

I'm not entirely sure when a new Silex V3 update is implemented, so I'm not aware if I'm using an old version. For inexperienced users, it might be confusing to have to clear the cache to use the latest version of V3.

Just a suggestion 🙂.

**Comment by lexoyo** [comment]
Yes that is true
There is a similar mechanism in v2
At least the user should know what's new in the version right ? Maybe since last time they opened silex? Or maybe use the version used to save the website

**Comment by GetLikha** [comment]
Yeah a sort of quick notification or maybe a "What's New?" panel in the homepage of V3 would be a nice feature to implement, then an update button in it too.







---- On Fri, 22 Sep 2023 08:00:22 -0300 Alex Hoyau ***@***.***> wrote ---





Yes that is true
 There is a similar mechanism in v2
 At least the user should know what's new in the version right ? Maybe since last time they opened silex? Or maybe use the version used to save the website

**Comment by lexoyo** [comment]
The update button would mean that there is a mechanism of pwa
Pwa means thet the whole front end is stored in the user's browser. Then it makes sense to have an update buttton. But in the case of silex i didn't implement the pwa mechanism, just a simple cache. So for now the update button would be useless as the app is updated automatically

**Comment by GetLikha** [comment]
> The update button would mean that there is a mechanism of pwa Pwa means thet the whole front end is stored in the user's browser. Then it makes sense to have an update buttton. But in the case of silex i didn't implement the pwa mechanism, just a simple cache. So for now the update button would be useless as the app is updated automatically

Ahh okay got it

![image](https://github.com/silexlabs/Silex/assets/132113888/4215ae63-72b8-4da9-a883-753d037c1637)
What I meant btw

But thanks for the clarification
